tidb-server/main.go

```diff
@@ -154,6 +156,13 @@ func main() {
 	os.Exit(0)
 }
 
+func syncLog() {
+	if err := log.Sync(); err != nil {
+		fmt.Fprintln(os.Stderr, "sync log err:", err)
+		os.Exit(1)
```

Comment:
after fix `Exit(0)`,  maybe we should ignore `log.Sync()` error here, in my linux laptop alway got:

```
2019/02/14 20:30:24.625 ddl.go:453: [info] [ddl] closing DDL:0cc14b05-77e7-4c7f-82da-80cb36a7a85c takes time 298.439µs
2019/02/14 20:30:24.625 ddl.go:376: [info] [ddl] stop DDL:0cc14b05-77e7-4c7f-82da-80cb36a7a85c
2019/02/14 20:30:24.625 domain.go:536: [info] [domain] close
sync log err: sync /dev/stdout: invalid argument
```

because linux stdout doesn't support `fsync`, zap also meet same problem https://github.com/uber-go/zap/pull/347 https://github.com/uber-go/zap/issues/370

it works well when using filesystem, but stdout or stderr always failure ...  I'm not confirm welther ignore it too - -?

Reply:
In most of the production environment, stdout is redirected to a log file, how about keeping the implementation, i.e. sync log, print the sync error to stderr?

Reply:
We have an inspection mechanism that can't ignore error detection. I will keep the current implementation first. If there are better suggestions, I will modify it.